src/lqcut.jl

```diff
@@ -52,7 +55,7 @@ function lq_drift_time_correction(
         drift_start = drift_prestats.peak_pos - prehist_sigma * drift_prestats.peak_sigma
         drift_stop = drift_prestats.peak_pos + prehist_sigma * drift_prestats.peak_sigma
         
-        drift_edges = range(drift_start, stop=drift_stop, length=ideal_length)
+        drift_edges = range(drift_start, stop=drift_stop, length=100)
```

Comment:
Could you elaborate on choosing a fixed value of  `100` here?

Reply:
I get an error with the fit. I need to check in more detail what happens and why it happens. That didn't happen when i was fitting a whole partition(instead of like now only one run). So just a tempory fix till i (hopefully) fix it properly. The error is:

caused by: Error in drift time correction: BoundsError([20, 41, 27, 43, 42, 44, 22], (0:7,))
Stacktrace:
 [1] macro expansion
   @ /mnt/artemis02/users/gieb/MPP_Code/legend-julia-dataflow/processors/a_process_lq.jl:126 [inlined]
 [2] macro expansion
   @ ~/.julia/packages/ProgressMeter/kVZZH/src/ProgressMeter.jl:1008 [inlined]
 [3] top-level scope
   @ /mnt/artemis02/users/gieb/MPP_Code/legend-julia-dataflow/processors/a_process_lq.jl:113

caused by: BoundsError: attempt to access 7-element Vector{Int64} at index [0:7]
Stacktrace:
  [1] throw_boundserror(A::Vector{Int64}, I::Tuple{UnitRange{Int64}})
    @ Base ./abstractarray.jl:737
  [2] checkbounds
    @ ./abstractarray.jl:702 [inlined]
  [3] view
    @ ./subarray.jl:184 [inlined]
  [4] estimate_single_peak_stats_th228(h::Histogram{Int64, 1, Tuple{StepRangeLen{…}}})
    @ LegendSpecFits /mnt/artemis02/users/gieb/MPP_Code/SpecFits/src/peakstats.jl:59
  [5] fit_binned_trunc_gauss(h_nocut::Histogram{…}, cuts::@NamedTuple{…}; uncertainty::Bool)
    @ LegendSpecFits /mnt/artemis02/users/gieb/MPP_Code/SpecFits/src/singlefit.jl:360
  [6] fit_binned_trunc_gauss
    @ /mnt/artemis02/users/gieb/MPP_Code/SpecFits/src/singlefit.jl:351 [inlined]
  [7] fit_binned_trunc_gauss(h_nocut::Histogram{Int64, 1, Tuple{StepRangeLen{…}}})
    @ LegendSpecFits /mnt/artemis02/users/gieb/MPP_Code/SpecFits/src/singlefit.jl:351
  [8] lq_drift_time_correction(lq_norm::Vector{…}, dt_eff::Vector{…}, e_cal::Vector{…}, DEP_μ::Quantity{…}, DEP_σ::Quantity{…}; DEP_edgesigma::Float64, mode::Symbol, lower_exclusion::Float64, upper_exclusion::Float64, drift_cutoff_sigma::Float64, prehist_sigma::Float64, e_expression::String)
    @ LegendSpecFits /mnt/artemis02/users/gieb/MPP_Code/SpecFits/src/lqcut.jl:61
  [9] macro expansion
    @ /mnt/artemis02/users/gieb/MPP_Code/legend-julia-dataflow/processors/a_process_lq.jl:123 [inlined]
 [10] macro expansion
    @ ~/.julia/packages/ProgressMeter/kVZZH/src/ProgressMeter.jl:1008 [inlined]
 [11] top-level scope
    @ /mnt/artemis02/users/gieb/MPP_Code/legend-julia-dataflow/processors/a_process_lq.jl:113
Some type information was truncated. Use `show(err)` to see complete types.

Reply:
The error message shows exactly where this comes from:

https://github.com/DaGeibl/LegendSpecFits.jl/blob/954d96aad356628db8da468bc748541bb24997b6/src/peakstats.jl#L58

as there is no entry more than 15 sigma away from the center position, the index defaults to length(array)-7. as your array only has 7 entries, it tries to access the array at index 0, which errors.